cwms-data-api/src/main/java/cwms/cda/data/dao/StreamDao.java

```diff
@@ -1,130 +1,257 @@
+     * @return a list of streams
+     */
+    public List<Stream> retrieveStreams(String officeIdMask, String streamIdMask) {
+        return connectionResult(dsl, conn -> {
+            setOffice(conn, officeIdMask);
+            ResultSet streams = CWMS_STREAM_PACKAGE.call_CAT_STREAMS(DSL.using(conn).configuration(), streamIdMask,
+                    DB_STATION_UNITS, null, null, null,
+                    null, null, null, null,
+                    null, null, null, null, null,
+                    null, null, officeIdMask)
+                    .intoResultSet();
+            return buildStreamListFromResultSet(streams);
         });
     }
 
-    private Set<StreamLocation> getStreamLocationsOnStream(String streamId, String unitSystem,
-                                                           String officeId) throws SQLException {
-        StreamLocationDao streamLocationDao = new StreamLocationDao(dsl);
-        return streamLocationDao.getStreamLocations(streamId, unitSystem, officeId);
+    /**
+     * Retrieve a specific stream
+     * @param streamId - the id of the stream
+     * @param officeId - the office id
+     * @return the stream
+     */
+    public Stream retrieveStream(String officeId, String streamId) {
+        return connectionResult(dsl, conn -> {
+            setOffice(conn, officeId);
+            STREAM_T streamT = CWMS_STREAM_PACKAGE.call_RETRIEVE_STREAM_F(DSL.using(conn).configuration(), streamId, DB_STATION_UNITS, officeId);
+            if (streamT == null) {
+                throw new NotFoundException("Stream: " + officeId + "." + streamId + " not found");
+            }
+            return fromJooqStream(streamT);
+        });
     }
 
-    private Set<StreamReach> getReaches(String streamId, String officeId) throws SQLException {
-        StreamReachDao streamReachDao = new StreamReachDao(dsl);
-        return streamReachDao.getReachesOnStream(streamId, officeId);
+    /**
+     * Store a stream
+     * @param stream - the stream to store
+     * @param failIfExists - if true, fail if the stream already exists
+     */
+    public void storeStream(Stream stream, boolean failIfExists) {
+        connectionResult(dsl, conn -> {
+            setOffice(conn, stream.getOfficeId());
+            String failsIfExistsStr = OracleTypeMap.formatBool(failIfExists);
+            String ignoreNullsStr = OracleTypeMap.formatBool(true);
+            String startsDownstream = OracleTypeMap.formatBool(stream.getStartsDownstream());
+            String stationUnits = getStationUnits(stream);
+            String flowIntoStream = null;
+            Double flowIntoStation = null;
+            String flowIntoBank = null;
+            String divertFromStream = null;
+            Double divertFromStation = null;
+            String divertFromBank = null;
+            if(stream.getFlowsIntoStreamNode() != null) {
+                flowIntoStream = stream.getFlowsIntoStreamNode().getStreamId().getName();
```

Comment:
readability: I'd extract `stream.getFlowsIntoStreamNode()` to a variable

Reply:
similarly to the other condition blocks below

Reply:
extracted to variable for both flowsInto and divertsFrom